aten/src/ATen/core/ivalue.h

```diff
@@ -623,4 +626,27 @@ inline optional<T> IValue::toOptional() {
   return this->to<T>();
 }
 
+inline bool IValue::isSameIdentity(IValue& rhs) {
+  // we choose to not use memcmp for payload check due to potenntial random padding characters on union type
```

Comment:
This logic is still a decent interpretation of `is` but will be stabler over time as we add new IValue variants:

```
return is_intrusive_ptr && rhs.is_intrusive_ptr && this->payload.as_intrusive_ptr == rhs.payload.as_intrusive_ptr;
```

Reply:
Oh, crap. That doesn't work for none. It might need to handle None is None as a special case.

Reply:
```(tag == rhs.tag && isNone() || s_intrusive_ptr && rhs.is_intrusive_ptr && this->payload.as_intrusive_ptr == rhs.payload.as_intrusive_ptr)```

Reply:
Oh good catch... I thought I added a none case in the below `else if` branch, turns out I forgot it after moved from memcmp the payloads.